packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -354,13 +369,13 @@ export default class ScriptTransformer {
   }
 
   transform(
-    filename: Path,
+    filename: Config.Path,
     options: Options,
     fileSource?: string,
-  ): TransformResult {
+  ): TransformResult | string {
     let scriptCacheKey = null;
     let instrument = false;
-    let result = '';
+    let result: TransformResult | string | undefined = '';
```

Comment:
Can it be really undefined?

Reply:
Changed to `this._cache.transformedFiles.get(scriptCacheKey) || ''` below